Fabric.Authorization.Domain/Services/GroupMigratorService.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Fabric.Authorization.Domain.Models;
+using Fabric.Authorization.Domain.Stores;
+using Serilog;
+
+namespace Fabric.Authorization.Domain.Services
+{
+    public class GroupMigratorService
+    {
+        private readonly IGroupStore _groupStore;
+        private readonly ILogger _logger;
+
+        public GroupMigratorService(
+            IGroupStore groupStore,
+            ILogger logger)
+        {
+            _groupStore = groupStore ?? throw new ArgumentNullException(nameof(groupStore));
+            _logger = logger;
+        }
+
+        public async Task<GroupMigrationResult> MigrateDuplicateGroups()
+        {
+            var groups = (await _groupStore.GetAll()).ToList();
```

Comment:
Should we only get non deleted groups? Might be weird if we merge groups when a groups is deleted...along those lines, will a deleted group contribute to the root cause of the original error?